tests/test_async_recorder.py

```diff
@@ -43,6 +44,28 @@ async def test_capture(loop):
     assert platform.python_implementation() == service.get('runtime')
     assert platform.python_version() == service.get('runtime_version')
 
+async def test_concurrent_calls(loop):
+    xray_recorder.configure(service='test', sampling=False, context=AsyncContext(loop=loop))
+    async with xray_recorder.in_segment_async('segment') as segment:
+        global counter
+        counter = 0
+        total_tasks = 10
+        event = asyncio.Event()
+        async def assert_task():
+            async with xray_recorder.in_subsegment_async('segment') as subsegment:
+                global counter
+                counter += 1
+                # Ensure that the task subsegments overlap
+                if counter < total_tasks:
+                    await event.wait()
+                else:
+                    event.set()
+                return subsegment.parent_id
+        tasks = [assert_task() for task in range(total_tasks)]
+        results = await asyncio.gather(*tasks)
+        for result in results:
+            assert result == segment.id
```

Comment:
Just a suggestion on naming:

```suggestion
        subsegs_parent_ids = await asyncio.gather(*tasks)
        for subseg_parent_id in subsegs_parent_ids:
            assert subseg_parent_id == segment.id
```